spec/models/framework/definition/language_spec.rb

```diff
@@ -91,6 +91,7 @@
 
           it 'is assumed to be present and a valid URN' do
             expect(invoice_class).to have_field('Customer URN')
+              .with_activemodel_type(:integer)
               .validated_by(:presence, :urn)
```

Comment:
Integers don't need a `presence` validator, as an empty field is not a number (optional values need `allow_nil: true`).

Reply:
I can see why in the case of URNs it isn't needed so we can remove it. 

I wasn't sure I understood "Integers don't need a `presence` validator, as an empty field is not a number (optional values need `allow_nil: true`)." so I ran the following:

```ruby
require 'active_model'

class SomeClass
  include ActiveModel::Attributes
  include ActiveModel::Validations

  attribute :some_int, :integer
  validates :some_int, presence: true
end

instance = SomeClass.new
[nil, '', '1', 1, 'not a number'].each do |int|
  instance.some_int = int
  instance.validate
  value_type = "#{int}:#{int.class}"
  puts "#{value_type.ljust(20)} results in a value of #{instance.some_int}, valid? == #{instance.valid?}"
end
```

the output of which is

```
:NilClass            results in a value of , valid? == false
:String              results in a value of , valid? == false
1:String             results in a value of 1, valid? == true
1:Integer            results in a value of 1, valid? == true
not a number:String  results in a value of 0, valid? == true
```

Commenting out `validates :some_int, presence: true`, the output is

```
:NilClass            results in a value of , valid? == true
:String              results in a value of , valid? == true
1:String             results in a value of 1, valid? == true
1:Integer            results in a value of 1, valid? == true
not a number:String  results in a value of 0, valid? == true
```

so it looks like saying something is an `:integer` attribute doesn't imply any validation, it just performs a different type cast when the value is set: 

- `instance.some_int = 'not a number'` nets you a value of 0, but
- `instance.some_int = nil` or `instance.some_int = ''` leaves the value `nil` while triggering the `presence` validation.

Do you mean integers – as we implement them in the current defs – as `:string` with an `ingested_numericality` validator? `allow_nil` has to be used in conjunction with at least one other validator or it will `raise ArgumentError, "You need to supply at least one validation"`